internal/api/core/task.go

```diff
@@ -16,29 +16,40 @@ import (
 // GetTask gets a task - currently only associated with kubernetes 'tasks'.
 func (cc *Controller) GetTask(c *gin.Context) {
 	id := c.Param("id")
+	task := clouddriver.NewDefaultTask(id)
 	manifests := []map[string]interface{}{}
 
 	resources, err := cc.SQLClient.ListKubernetesResourcesByTaskID(id)
 	if err != nil {
-		clouddriver.Error(c, http.StatusBadRequest, err)
```

Comment:
Will we need the extra detail that `clouddriver.Error` provides? I imagine that you put it there for a reason. It was used a lot in this function. My reasoning is that we probably don't because:
1. you took it away
2. because the gin error should have enough detail.

Just wanted to make sure it wasn't an oversight.

```
// Error attaches a given Go error to a gin context and sets its type to public.
func Error(c *gin.Context, status int, err error) {
	pc, fn, ln, _ := runtime.Caller(1)
	m := ErrorMeta{
		FuncName: runtime.FuncForPC(pc).Name(),
		FileName: fn,
		GUID:     uuid.New().String(),
		LineNum:  ln,
	}

	c.Status(status)
	_ = c.Error(err).SetType(gin.ErrorTypePublic).SetMeta(m)
}

```

Reply:
`clouddriver.Error` is used for the following
- to define some metadata about where the error came from
- to attach an error to the gin context

We handle "Clouddriver Errors" using the `HandleError` Middleware [here](https://github.com/homedepot/go-clouddriver/blob/master/internal/middleware/error.go). This middleware is responsible for actually writing to the response writer with the error and status code. It also logs the error if it's a HTTP status of 500+.

These are nice things to have, but this middleware specifically writes a "Clouddriver Error", which looks like:
```json6
{
  "error": "Internal Server Error (error ID: bee2337f-94fe-4fa6-86b2-7417abc59df9)",
  "message": "some error message",
  "status": 500,
  "timestamp": 1647384052654
}
```
The above error won't work for the Task API endpoint, and it's actually the reason I opened this PR. The Monitor stage task is expecting an object that contains information about the operation at `$.status`, not an integer, so it throws a null pointer in Orca.

Considering we're wrapping these Task API errors in some unique text it should be easy to see where an error originated, so probably not important to have all the extra nice stuff for the Task API.